avm/src/daemon/daemon.rs

```diff
@@ -213,14 +225,74 @@ async fn get_v1_stats() -> VMStatsV1 {
 // returns cluster delta
 async fn post_v1_stats(cluster_id: &str, deploy_token: &str) -> String {
   let vm_stats = get_v1_stats().await;
-  let stats_body = json!({
+  let mut stats_body = json!({
     "deployToken": deploy_token,
     "vmStats": vm_stats,
     "clusterId": cluster_id,
   });
+  let cluster_secret = CLUSTER_SECRET.get().unwrap();
+  if let Some(cluster_secret) = cluster_secret.as_ref() {
+    stats_body
+      .as_object_mut()
+      .unwrap()
+      .insert("clusterSecret".to_string(), json!(cluster_secret));
+  }
   post_v1("stats", stats_body).await
 }
 
+async fn control_port(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+  let cluster_secret = CLUSTER_SECRET.get().unwrap();
+  if cluster_secret.is_some() && !req.headers().contains_key(cluster_secret.as_ref().unwrap()) {
+    // If this control port is guarded by a secret string, make sure there's a header with that
+    // secret as the key (we don't care about the value) and abort otherwise
```

Comment:
so the key to the header will be the `CLUSTER_SECRET` and the value empty?

Reply:
Sorta. I think you have to set it to *something* so I set it to `true` in my test above. We could eventually rotate the secret, for instance? We could have it be `CURRENT_SECRET: CURRENT_SECRET` normally and `CURRENT_SECRET: REPLACEMENT_SECRET` if/when we want to rotate the secret out.

That will require replacing `OnceCell` with something else, but something to think about on the security front. The shorter-lived a secret, the harder it is to do anything with it.

Reply:
it still works with `OnceCell` <https://docs.rs/once_cell/1.7.2/once_cell/sync/struct.OnceCell.html#method.take>

Reply:
> Sorta. I think you have to set it to _something_ so I set it to `true` in my test above. We could eventually rotate the secret, for instance? We could have it be `CURRENT_SECRET: CURRENT_SECRET` normally and `CURRENT_SECRET: REPLACEMENT_SECRET` if/when we want to rotate the secret out.
> 
> That will require replacing `OnceCell` with something else, but something to think about on the security front. The shorter-lived a secret, the harder it is to do anything with it.

Though when we want to dig into secret rotation, we should think hard on the implications of the API. It would be useful to be able to obfuscate *how* the rotation works, too, if that is somehow possible.

Reply:
> it still works with `OnceCell` https://docs.rs/once_cell/1.7.2/once_cell/sync/struct.OnceCell.html#method.take

For certain definitions of "once," it seems. 🤨